### ISSUE TITLE
explore adding a bot to test changes against the Flutter codebase

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/linter#1205]
We should explore adding a bot to test changes against the Flutter codebase.

**Comment by bwilkerson** [comment]
Would this be different than the bot used to test analyzer against Flutter?

**Comment by devoncarew** [comment]
Yes - it would run for PRs into the linter, and would show up here: https://travis-ci.org/dart-lang/linter.

Testing analyzer against Flutter would likely be done via the chromium bot infrastructure (https://github.com/dart-lang/sdk/issues/34739).

**Comment by srawlins** [comment]
It looks like we could run a GitHub action [only when certain labels are applied](https://stackoverflow.com/questions/62325286/run-github-actions-when-pull-requests-have-a-specific-label), and so we could make this an optional CI job, for flutter-related rules. CC @pq

**Comment by devoncarew** [comment]
This workflow config - from dart-lang/markdown - might be a useful template: https://github.com/dart-lang/markdown/blob/master/.github/workflows/flutter_markdown.yaml.

**Comment by srawlins** [comment]
I'm pretty sure we would need to compile a Dart SDK with the local linter changes, and use that SDK in `flutter analyze ...`
